packages/roosterjs-editor-core/lib/coreApi/createPasteFragment.ts

```diff
@@ -60,5 +67,18 @@ function createBeforePasteEvent(core: EditorCore, clipboardData: ClipboardData):
         htmlBefore: '',
         htmlAfter: '',
         htmlAttributes: {},
+        pasteType: pasteType,
     };
 }
+
+function getPasteType(pasteAsText: boolean, applyCurrentStyle: boolean, pasteAsImage: boolean) {
+    if (!pasteAsImage && !applyCurrentStyle && !pasteAsText) {
```

Comment:
Let's reorder this check, how about

if (pasteAsText) { return AsPlainText } 
else if (applyCurrentStyle) { ... }
else if (pasteAsImage) { ... }
else { return Normal}